### PR TITLE
debugedit-unstable: init at e04296ddf / 2021-07-05

### PR DESCRIPTION
**Pull request by deliciouslytyped**
###### Motivation for this change
Adds a tool for a limited set of manipulation operations for file paths in debug info sections.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build debugedit-unstable

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#TODO@deliciouslytyped The tool seems to unnecessarily force mutable access for the debugedit `-l` feature
+{fetchgit, lib, stdenv, autoreconfHook, pkgconfig, elfutils, help2man, util-linux}:
+stdenv.mkDerivation {
+  name = "debugedit";
+  version = "unstable-2021-07-05";
+
+  buildInputs = [ autoreconfHook pkgconfig elfutils help2man ];
```
Comment (symphorien) [inline pkgs/development/tools/misc/debugedit/default.nix:7]: Probably only elfutil is a buildInput, the other ones are nativeBuildInputs. If unsure, add strictDeps = true.

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  preBuild = ''
+    patchShebangs scripts/find-debuginfo.in
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "debugedit provides programs and scripts for creating debuginfo and source file distributions, collect build-ids and rewrite source paths in DWARF data for debugging, tracing and profiling.";
+    homepage = "https://sourceware.org/debugedit/";
+    license = licenses.gpl2;
```
Comment (symphorien) [inline pkgs/development/tools/misc/debugedit/default.nix:25]: please disambiguate into gpl2Only or gpl2Plus.

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#TODO@deliciouslytyped The tool seems to unnecessarily force mutable access for the debugedit `-l` feature
+{fetchgit, lib, stdenv, autoreconfHook, pkgconfig, elfutils, help2man, util-linux}:
```
Comment (vcunat) [inline pkgs/development/tools/misc/debugedit/default.nix:2]: `pkgconfig` is only an alias, so use `pkg-config` instead.  That's why OfBorg complains.

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  preBuild = ''
+    patchShebangs scripts/find-debuginfo.in
+  '';
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "debugedit provides programs and scripts for creating debuginfo and source file distributions, collect build-ids and rewrite source paths in DWARF data for debugging, tracing and profiling.";
```
Comment (symphorien) [inline pkgs/development/tools/misc/debugedit/default.nix:24]: According to contributing.md:

> meta.description should:
>    Be capitalized.
>    Not start with the package name.
>    Not have a period at the end.

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "https://sourceware.org/debugedit/";
+    license = licenses.gpl2Plus;
```
Comment (symphorien) [inline pkgs/development/tools/misc/debugedit/default.nix:26]: Homepage says
> The debuginfo project as a whole can be redistributed under the GPLv3+. Some individual source files may also be redistributed under the GPLv2+ or under the LGPLv2+. See the headers of the individual files to learn which licenses apply. All licenses used are upward compatible with GPLv3+.

File: pkgs/development/tools/misc/debugedit/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+#TODO@deliciouslytyped The tool seems to unnecessarily force mutable access for the debugedit `-l` feature
+{fetchgit, lib, stdenv, autoreconfHook, pkg-config, elfutils, help2man, util-linux}:
+stdenv.mkDerivation {
+  name = "debugedit";
+  version = "unstable-2021-07-05";
```
Comment (symphorien) [inline pkgs/development/tools/misc/debugedit/default.nix:5]: There was a release in june: https://sourceware.org/git/?p=debugedit.git;a=tag;h=97ee9665f8d401d52340845749728ebb7c482a79

any reason to use current master?

Comment (deliciouslytyped) [reply pkgs/development/tools/misc/debugedit/default.nix:5]: I intend to contact upstream about some usability fixes that I would expect to get in relatively quickly.

File: pkgs/development/tools/misc/debugedit/default.nix (hunk 5 above)
Comment (deliciouslytyped) [reply pkgs/development/tools/misc/debugedit/default.nix:26]: Aha, I only looked at a source header. I guess GPL3+ then?

File: pkgs/development/tools/misc/debugedit/default.nix (hunk 4 above)
Comment (deliciouslytyped) [reply pkgs/development/tools/misc/debugedit/default.nix:24]: I copied the description straight from the first visible thing on the website. If we want to diverge from that I'll just remove the first word, capitaiize the second, and remove the period.Weird to ask for capitalization but no punctuation, but that's out of scope here *shrug*.

Comment (symphorien) [reply pkgs/development/tools/misc/debugedit/default.nix:24]: Please do.

File: pkgs/development/tools/misc/debugedit/default.nix (hunk 5 above)
Comment (symphorien) [reply pkgs/development/tools/misc/debugedit/default.nix:26]: Unless you want to check *all* source files, I suppose it's better to trust the author :)

File: pkgs/development/tools/misc/debugedit/default.nix (hunk 6 above)
Comment (symphorien) [reply pkgs/development/tools/misc/debugedit/default.nix:5]: fine then :)

File: pkgs/development/tools/misc/debugedit/default.nix (hunk 5 above)
Comment (deliciouslytyped) [reply pkgs/development/tools/misc/debugedit/default.nix]: Yes, but I was insufficiently observant to notice that on the home page. :)

**Pull request merged**